tests/evpn_tunnel.py

```diff
@@ -0,0 +1,1143 @@
+
+        self.helper.delete_entry_pst(app_db, "ROUTE_TABLE", "%s:%s" % (vrf_name, prefix))
+
+        time.sleep(2)
+
+    def create_vrf_routes_ecmp(self, dvs, prefix, vrf_name, ecmp_nexthop_attributes):
+        app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+
+        self.helper.create_entry_pst(
+            app_db,
+            "ROUTE_TABLE",  "%s:%s" % (vrf_name, prefix),
+            ecmp_nexthop_attributes,
+        )
+
+        time.sleep(2)
+
+    def create_vlan(self, dvs, vlan_name, vlan_ids):
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+        vlan_id = vlan_name[4:]
+
+        # create vlan
+        self.helper.create_entry_tbl(
+            conf_db,
+            "VLAN",  vlan_name,
+            [
+              ("vlanid", vlan_id),
+           ],
+        )
+
+        time.sleep(1)
+
+        vlan_oid = self.helper.get_created_entry(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_VLAN", vlan_ids)
+
+        self.helper.check_object(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_VLAN", vlan_oid,
+                        {
+                            "SAI_VLAN_ATTR_VLAN_ID": vlan_id,
+                        }
+                    )
+
+        return vlan_oid
+
+    def remove_vlan(self, dvs, vlan):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(conf_db, "VLAN")
+        tbl._del("Vlan" + vlan)
+        time.sleep(1)
+
+    def create_vlan_member(self, dvs, vlan, interface, tagging_mode="untagged"):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(conf_db, "VLAN_MEMBER")
+        fvs = swsscommon.FieldValuePairs([("tagging_mode", tagging_mode)])
+        tbl.set("Vlan" + vlan + "|" + interface, fvs)
+        time.sleep(1)
+
+    def remove_vlan_member(self, dvs, vlan, interface):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(conf_db, "VLAN_MEMBER")
+        tbl._del("Vlan" + vlan + "|" + interface)
+        time.sleep(1)
+
+
+    def create_vlan_interface(self, dvs, vlan_name, ifname, vrf_name, ipaddr):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+        # create a vlan member in config db
+        self.helper.create_entry_tbl(
+            conf_db,
+            "VLAN_MEMBER",  "%s|%s" % (vlan_name, ifname),
+            [
+              ("tagging_mode", "untagged"),
+            ],
+        )
+
+        time.sleep(1)
+
+        # create vlan interface in config db
+        self.helper.create_entry_tbl(
+            conf_db,
+            "VLAN_INTERFACE",  vlan_name,
+            [
+              ("vrf_name", vrf_name),
+            ],
+        )
+
+        app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+        self.helper.create_entry_pst(
```

Comment:
Could you please elaborate a bit why do we need to create this entry in APP_DB?
As far as I know to create a RIF it is enough to apply 2 entries to CFG_DB, like this:
```
    "VLAN_INTERFACE": {
        "Vlan1000": {},
        "Vlan1000|192.168.0.1/21": {}
    },
```

Reply:
This was legacy code and I merely refactored it. I am not sure why it was defined like this in the first place.